--- a/config/install/hdf5-config.cmake.in
+++ b/config/install/hdf5-config.cmake.in
@@ -124,7 +124,7 @@
   endif ()
 endif ()

-if (${HDF5_PACKAGE_NAME}_PROVIDES_ZLIB_SUPPORT AND NOT @ZLIB_USE_EXTERNAL@)
+if (${HDF5_PACKAGE_NAME}_PROVIDES_ZLIB_SUPPORT AND ${HDF5_PACKAGE_NAME}_PROVIDES_STATIC_LIBS AND NOT @ZLIB_USE_EXTERNAL@)
   if (NOT @HDF5_USE_ZLIB_NG@)
     if (@HDF5_MODULE_MODE_ZLIB@)
       # Expect that the default shared library is expected with FindZLIB.cmake
@@ -139,7 +139,7 @@

 # If szip/libaec support was enabled and libaec was found on the system at build
 # time, locate libaec to propagate the dependency
-if (${HDF5_PACKAGE_NAME}_PROVIDES_SZIP_SUPPORT AND NOT @SZIP_USE_EXTERNAL@)
+if (${HDF5_PACKAGE_NAME}_PROVIDES_SZIP_SUPPORT AND ${HDF5_PACKAGE_NAME}_PROVIDES_STATIC_LIBS AND NOT @SZIP_USE_EXTERNAL@)
   find_dependency (@LIBAEC_PACKAGE_NAME@ OPTIONAL_COMPONENTS @LIBAEC_SEARCH_TYPE@)
 endif ()
